### PR TITLE
:seedling: provide a normalize library to generate scores for AddOnPlacementScore

### PR DESCRIPTION
**Pull request by haoqing0110**
## Summary
Ref: https://github.com/open-cluster-management-io/ocm/issues/68

## Related issue(s)

Fixes #

### CODE REVIEW

File: pkg/apis/cluster/v1alpha1/score.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+		min: min,
+		max: max,
+	}
+}
+
+// Normalize normalizes a given value to the range -100 to 100 based on the min and max values.
+func (s *scoreNormalizer) Normalize(value float64) (score int64, err error) {
+	if value > s.max {
+		// If the value exceeds the maximum, set score to MaxScore.
+		score = int64(MaxScore)
```
Comment (qiujian16) [inline pkg/apis/cluster/v1alpha1/score.go:32]: why we need to define MinScore as float and convert to int here?

Comment (haoqing0110) [reply pkg/apis/cluster/v1alpha1/score.go:32]: the MinScore and MaxScore should also used to normalize the score when s.min < value < s.max, just update the formula on line 38.

File: pkg/apis/cluster/v1alpha1/score.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+		score = MaxScore
+		// If the value is less than or equal to the minimum, set score to MinScore.
+	} else if value <= s.min {
+		score = MinScore
+	} else {
+		// Otherwise, normalize the value to the range -100 to 100.
+		score = (int32)((MaxScore-MinScore)*(value-s.min)/(s.max-s.min) + MinScore)
+	}
+
+	klog.Infof("value = %v, min = %v, max = %v, score = %v", value, s.min, s.max, score)
```
Comment (qiujian16) [inline pkg/apis/cluster/v1alpha1/score.go:41]: V(2)

Comment (haoqing0110) [reply pkg/apis/cluster/v1alpha1/score.go]: fixed.

**Pull request merged**